mlflow/store/tracking/rest_store.py

```diff
@@ -250,10 +250,22 @@ def get_metric_history(self, run_id, metric_key):
         :return: A list of :py:class:`mlflow.entities.Metric` entities if logged, else empty list
         """
         req_body = message_to_json(
-            GetMetricHistory(run_uuid=run_id, run_id=run_id, metric_key=metric_key)
+            GetMetricHistory(
+                run_uuid=run_id,
+                run_id=run_id,
+                metric_key=metric_key,
+                max_results=max_results,
+                page_token=page_token,
+            )
         )
         response_proto = self._call_endpoint(GetMetricHistory, req_body)
-        return [Metric.from_proto(metric) for metric in response_proto.metrics]
+
+        metric_history = [Metric.from_proto(metric) for metric in response_proto.metrics]
+        if response_proto.next_page_token:
+            next_page_token = response_proto.next_page_token
+            return metric_history, next_page_token
+        else:
+            return metric_history
```

Comment:
this logic needs to change to support backwards compatibility. Pushing change soon.